Hyperion/Extensions/TypeEx.cs

```diff
@@ -200,24 +203,27 @@ public static int GetTypeSize(this Type type)
 
         public static string GetShortAssemblyQualifiedName(this Type type)
```

Comment:
@Horusiath I was looking into this. There is an issue with this optimization. For ObjectSerializer this idea works fine, but TypeSerializer has it's own way to serialize values directy to the stream. I tried replacing the StringSerializer for the ByteArraySerializer and the tests didn't pass. We would need 2 caches one for each format, or change something else to make them equal.

Another thing is that since GetShortAssemblyQualifiedName is recursive, caching itself helps a little resolving names the first time. Then again, we could add second cache, but increase memory usage. Not sure if there would be significant gains.

Using the cache should make the protocol at least as fast as it was before. Worst case, a dictionary + lock could be faster in some scenarios. 

This is getting complicated... =) I believe it's better to merge this at is and try to optimize this later.